core/src/main/java/fj/Hash.java

```diff
@@ -262,6 +262,25 @@ public int hash(final A a) {
     });
   }
 
+  /**
+   * A hash instance for the {@link Zipper} type.
+   *
+   * @param ha A hash for the elements of the zipper.
+   * @return A hash instance for the {@link Zipper} type.
+   */
+  public static <A> Hash<Zipper<A>> zipperHash(final Hash<A> ha) {
+    return hash(as -> {
+        final int p = 419;
+        int r = 239;
+
+        r = p * r + streamHash(ha).hash(as.lefts());
```

Comment:
also probably best to save `streamHash(ha)` in a local variable.

Reply:
done